backend/src/common/types/repository.type.ts

```diff
@@ -1,9 +1,9 @@
 type Repository<T = unknown> = {
     find(query: Record<string, T>): Promise<T>;
-    findAll(): Promise<T[]>;
+    findAll(query: Record<string, T>): Promise<T[]>;
```

Comment:
I think we don't need query for findAll.  In repository we can write .query().execute() and it will return all data

Reply:
You are right